modules/module-postgres-storage/test/src/connection-report-storage.test.ts

```diff
@@ -0,0 +1,339 @@
+import { afterAll, beforeAll, describe, expect, it } from 'vitest';
+import { POSTGRES_REPORT_STORAGE_FACTORY } from './util.js';
+import { event_types } from '@powersync/service-types';
+
+function removeVolatileFields(sdks: event_types.ClientConnection[]): Partial<event_types.ClientConnection>[] {
+  return sdks.map((sdk) => {
+    const { id, disconnected_at, connected_at, jwt_exp, ...rest } = sdk;
+    return {
+      ...rest
+    };
+  });
+}
+
+describe('Connection report storage', async () => {
```

Comment:
Same as for MongoDB, tests should also specifically include a case where `client_id` is undefined.

Reply:
Additionally, is it feasible for these tests to primarily use the public storage APIs, rather than raw SQL? That way the tests could be shared between the Postgres and MongoDB implementations, potentially only having small number of db-specific tests if you do still need the lower-level access in some places.

Reply:
There should still be a user-agent in those cases, although it would likely be the generic platform-specific one.

I'd recommend using a blank string for the client_id if you want to avoid using `undefined`.